src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -7065,8 +7065,16 @@ public Tensor stft(long n_fft, long hop_length = -1, long win_length = -1, Tenso
             /// <returns></returns>
             public Tensor istft(long n_fft, long hop_length = -1, long win_length = -1, Tensor? window = null, bool center = true, bool normalized = false, bool? onesided = null, long length = -1, bool return_complex = false)
             {
+                var fft_size = shape[1];
                 IntPtr _window = (window is null) ? IntPtr.Zero : window.Handle;
-                bool _onesided = onesided.HasValue ? onesided.Value : !is_complex();
+
+                int _onesided = -1; // encoding of Python 'None'
+                if (onesided.HasValue) {
+                    _onesided = (onesided.Value ? 1 : 0);
+                } else if (n_fft != fft_size) {
+                    _onesided = 1;
+                }
```

Comment:
Probably these are handled in C++ code of PyTorch.

https://github.com/pytorch/pytorch/blob/4906a956181fb337e99860fa3e6071473c0e30e6/aten/src/ATen/native/SpectralOps.cpp#L1005

```
  const bool onesided = onesidedOpt.value_or(fft_size != n_fft);
```

Reply:
I've added hfft2/hfftn/ihfft2/ihfftn to 'torch' in this PR, too.